presto-main/src/main/java/com/facebook/presto/sql/planner/SymbolsExtractor.java

```diff
@@ -70,17 +78,34 @@ private SymbolsExtractor() {}
         return uniqueSymbols.build();
     }
 
+    public static Set<VariableReferenceExpression> extractUniqueVariable(PlanNode node, Lookup lookup, TypeProvider types)
+    {
+        ImmutableSet.Builder<VariableReferenceExpression> unique = ImmutableSet.builder();
+        extractExpressions(node, lookup).forEach(expression -> unique.addAll(extractUniqueVariableInternal(expression, types)));
+        return unique.build();
+    }
+
     public static Set<Symbol> extractUnique(Expression expression)
     {
         return ImmutableSet.copyOf(extractAll(expression));
     }
 
+    public static Set<VariableReferenceExpression> extractUniqueVariable(Expression expression, TypeProvider types)
+    {
+        return ImmutableSet.copyOf(extractAllVariable(expression, types));
+    }
+
     // TODO: return Set<VariableReferenceExpression>
```

Comment:
I assume ultimately `extractUniqueVariable` will be renamed to `extractUnique` to replace the symbol ones right?

Reply:
ya, i'll do more clean up after this PR. Just to unblock PlanNode to SPI first.